eppo_core/src/background/async_runtime.rs

```diff
@@ -0,0 +1,20 @@
+use std::future::Future;
+
+/// `AsyncRuntime` abstracts over various Rust's async runtimes with minimal interface. This is
+/// usually just a tokio Handle but Dart SDK in wasm uses a custom implementation.
```

Comment:
I did most of my reading around this change to understand it and got introduced to the `Send` interface, thanks! https://doc.rust-lang.org/nomicon/send-and-sync.html

Also needed help from AI to understand how it fit together with the wasm single threaded runtime. 

```
In the future, if more non-Send futures need handling, you may need a more generalized solution (e.g., perhaps relax the trait bound for wasm via conditional compilation, or provide a separate method for non-Send tasks). For now, the approach is correct and contained to config polling.
```

What do you imagine doing down the line to evolve this interface?

Reply:
Yeah, I don't really like this `AsyncRuntime` and I kinda used an escape hatch in configuration poller by invoking wasm spawn directly.

Having different definition for wasm is possible but it's going to be contagious as interfaces of `spawn_tracked`/`_untracked` would change as well. I don't think it's going to propagate much further though, so probably worth biting that bullet at some point.

I'm also not entirely sure what's multi-threading story for wasm is. It looks like there is some support for threading on some platforms and on web via [`SharedArrayBuffer`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer) (when cross-origin isolation is enabled).